### ISSUE TITLE
OptBucketer Index error

### ISSUE THREAD

**Issue by orchardbirds** [ing-bank/skorecard#33]
From the hackathon, trying to simply fit on the data

```python 

from skorecard.bucketers import OptimalBucketer

target = ['target']
features = [f for f in train.columns if f not in target]

X = train[features]
y = train[target]

X_train, X_valid, y_train, y_valid = train_test_split(X, y, test_size=0.25, random_state=411114)

bucketer = OptimalBucketer(max_n_bins=20, variables=features)
bucketer.fit(X_train, y_train)
```

outputs:

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-6-27fef8132e80> in <module>
     10 
     11 bucketer = OptimalBucketer(max_n_bins=20, variables=features, min_bin_size=0.001)
---> 12 bucketer.fit(X_train, y_train)

~/miniconda3/envs/skorecard_py37/lib/python3.7/site-packages/skorecard/bucketers/base_bucketer.py in fit(self, X, y)
    187             # This method is implemented by each bucketer
    188             assert isinstance(X_flt, pd.Series)
--> 189             splits, right = self._get_feature_splits(feature, X=X_flt, y=y_flt, X_unfiltered=X)
    190 
    191             # Deal with missing values

~/miniconda3/envs/skorecard_py37/lib/python3.7/site-packages/skorecard/bucketers/bucketers.py in _get_feature_splits(self, feature, X, y, X_unfiltered)
    154             **self.kwargs,
    155         )
--> 156         binner.fit(X.values, y)
    157 
    158         # Extract fitted boundaries

~/miniconda3/envs/skorecard_py37/lib/python3.7/site-packages/optbinning/binning/binning.py in fit(self, x, y, sample_weight, check_input)
    526             Fitted optimal binning.
    527         """
--> 528         return self._fit(x, y, sample_weight, check_input)
    529 
    530     def fit_transform(self, x, y, sample_weight=None, metric="woe",

~/miniconda3/envs/skorecard_py37/lib/python3.7/site-packages/optbinning/binning/binning.py in _fit(self, x, y, sample_weight, check_input)
    693             self.dtype, x, y, self.special_codes, self.cat_cutoff,
    694             self.user_splits, check_input, self.outlier_detector,
--> 695             self.outlier_params, None, None, self.class_weight, sample_weight)
    696 
    697         self._time_preprocessing = time.perf_counter() - time_preprocessing

~/miniconda3/envs/skorecard_py37/lib/python3.7/site-packages/optbinning/preprocessing.py in split_data(dtype, x, y, special_codes, cat_cutoff, user_splits, check_input, outlier_detector, outlier_params, fix_lb, fix_ub, class_weight, sample_weight)
    190         clean_mask = ~missing_mask
    191 
--> 192         x_clean = x[clean_mask]
    193         y_clean = y[clean_mask]
    194         x_missing = x[missing_mask]

IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed

```

**Comment by orchardbirds** [comment]
Same as https://github.com/ing-bank/skorecard/issues/34

**Comment by orchardbirds** [comment]
closed with #37
